### PR TITLE
Gas table update

### PR DESCRIPTION
**Pull request by pirapira**
This might fix or not #1247 (hope Github does not catch this).

It's not testable until https://github.com/ethereum/cpp-ethereum/issues/3504 is solved.  It might be proper software engineering if we solve that first.

### CODE REVIEW

**Comment by chriseth** [comment]
I think it suffices to use a recent ethereum version (i.e. if you want to run it on travis, you need to change something there) and then change `RPCSession::test_setChainParams` to include:

```
"params": {
...
		"homsteadForkBlock": "0x00",
		"EIP150ForkBlock": "0x00",
		"EIP158ForkBlock": "0x00",
...
}
```

**Comment by pirapira** [comment]
@chriseth that might work.  So I'm making the tests fail.

**Comment by pirapira** [comment]
https://github.com/ethereum/cpp-ethereum/pull/3514 needs to be solved before tests succeed.

File: Changelog.md [hunk 1]
```diff
@@ -29,6 +29,9 @@ Bugfixes:
  * Type checker: Allow multiple events of the same name (but with different arities or argument types)
  * Natspec parser: Fix error with ``@param`` parsing and whitespace.
 
+Bugfixes:
+ * Gas-estimator: reflect the most recent fee schedule.
```
Comment (axic) [inline Changelog.md]: Need to move this to the right version.

File: libevmasm/GasMeter.cpp [hunk 2]
```diff
@@ -148,6 +148,9 @@ GasMeter::GasConsumption GasMeter::estimateMax(AssemblyItem const& _item, bool _
 			}
 			break;
 		}
+		case Instruction::SELFDESTRUCT:
+			gas = GasCosts::selfdestructGas;
+			gas += GasCosts::callNewAccountGas; // We very rarely know whether the address exists.
 		case Instruction::CREATE:
```
Comment (axic) [inline libevmasm/GasMeter.cpp]: Isn't a break missing here?

Comment (pirapira) [reply libevmasm/GasMeter.cpp]: Yes, it's missing.

File: Changelog.md (hunk 1 above)
Comment (pirapira) [reply Changelog.md]: Yes, moved this.

**Comment by axic** [comment]
This might need #1664.

**Review by chriseth — APPROVED** [review]

**Comment by pirapira** [comment]
I cannot reproduce the test failure locally using soltest against `cpp-ethereum` `develop` 50fcad3c22149cea5c28ed07b73efebde820f8af

**Comment by pirapira** [comment]
Currently, the Travis error is not about the gas costs, but about `eth` killing blockchain.

```
 ⚡   17:24:27.416|<unknown>  Stop worker 310 ms

  ℹ  17:24:27.428|<unknown>  Killing blockchain & extras database (WithExisting::Kill).

  ℹ  17:24:27.447|<unknown>  Killing state database (WithExisting::Kill).

 ⚡   17:24:28.461|eth  Worker stopping 1045 ms

/home/travis/build/ethereum/solidity/test/RPCSession.cpp(128): fatal error in "failed_create": Timeout reading on IPC.
```

**Comment by chriseth** [comment]
The message about killing the blockchain is normal, the timeout is the actual error. "failed_create" is a transaction that takes very long to mine - perhaps we have to adjust the maximum time we wait for a mining response.

**Comment by axic** [comment]
`failed_create` tries to recurse 1023 times and then create a contract. Maybe the docker on travis is that slow. Increased timeout to 2mins for a test.

**Comment by pirapira** [comment]
I'm getting interested.

**Comment by chriseth** [comment]
Perhaps it is better to use a specific binary instead of docker. We only need to run the end to end tests on a single platform now, because we do bytecode comparison. We can use this binary here (I created it from cpp-ethereum/develop right now):

https://github.com/ethereum/cpp-ethereum/releases/download/solidityTester/eth

**Comment by pirapira** [comment]
Many tests fail for some reason.

**Comment by pirapira** [comment]
Many tests still fail for some reason.

**Comment by axic** [comment]
Must be rebased after JSON I/O was merged.

File: Changelog.md [hunk 3]
```diff
@@ -14,6 +14,7 @@ Bugfixes:
  * Compiler interface: Fix a bug where source indexes could be inconsistent between Solidity compiled
    with different compilers (clang vs. gcc) or compiler settings. The bug was visible in AST
    and source mappings.
+ * Gas-estimator: reflect the most recent fee schedule.
```
Comment (axic) [inline Changelog.md]: Probably this should be `Gas Estimator`?

**Pull request merged**